app/src/androidTest/java/ch/epfl/sweng/runpharaa/util/ViewUtils.java

```diff
@@ -52,4 +61,30 @@ public String getDescription() {
             }
         };
     }
+
+    static DataInteraction onPreferenceRow(Matcher<? extends Object> datamatcher) {
+
+        DataInteraction interaction = onData(datamatcher);
+
+        return interaction
+                .inAdapterView(Matchers.allOf(
+                        withId(android.R.id.list),
+                        not(withParent(withResName("headers")))));
+    }
+
+    static Matcher<View> withResName(final String resName) {
```

Comment:
What is the purpose of this function?

Reply:
It's used to test preferences because espresso couldn't find them when simply using `onView(withKey(...))`. Taken from here: https://stackoverflow.com/a/44262604

I will update the class to include more information on the methods.

Reply:
Ok thank you